packages/react-hooks/src/useOnChange.ts

```diff
@@ -0,0 +1,17 @@
+import { DependencyList } from 'react';
+import usePrevious from './usePrevious';
+
+/**
+ * Custom hook that triggers a callback function immediately when any of the dependencies change.
+ *
+ * @param callback - The function to be called when the dependencies change.
+ * @param deps - The list of dependencies to watch for changes.
+ */
+export function useOnChange(callback: () => void, deps: DependencyList): void {
+  const prevDeps = usePrevious(deps);
+  if (prevDeps === undefined || !deps.every((dep, i) => dep === prevDeps[i])) {
+    callback();
+  }
+}
```

Comment:
How is this different from `useEffect`? It's not immediate since `usePrevious` relies on `useEffect` to change its value. And `useEffect` already tracks the previous value for deps. I don't think this hook is necessary to replace the `useEffect` calls in `IrisGridModelUpdater`.

Reply:
When we were testing using the existing `useEffect` 's, the code was not working, which I believe was because they run after the render.

Reply:
This would call the callback during render, but `usePrevious` will trigger its updated value using `useEffect`. So I guess this works it just wouldn't have an updated `usePrevious` value if it ran twice in a render. So the callback may be called multiple times since `usePrevious` won't change until after `useEffect` has run.

If `useEffect` wasn't working, was there something changing the value mid-render that would cause it to not work?

Also, if you wanted true `onChange` and not just change from the last render if something interrupted and caused a 2nd render pass, then use a ref to track the prev deps and update it at the end of the hook

Reply:
Changing to a ref caused infinite loading for the table, so leaving `usePrevious` which uses `useEffect` is fine